src/BitcoinJsonRpc.ts

```diff
@@ -98,12 +98,36 @@ export default class BitcoinJsonRpc {
     }
   }
 
+
   public async sendRawTransaction(hex: string) {
     return this.cmdWithRetryAndDecode(decoders.SendRawTransactionResultDecoder, 'sendrawtransaction', hex);
   }
 
-  public async sendToAddress(address: string, amount: string) {
-    return this.cmdWithRetryAndDecode(decoders.SendToAddressResultDecoder, 'sendtoaddress', address, amount);
+  // https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/wallet/sendtoaddress/
+  public async sendToAddress(address: string, amount: string, comment?: string, commentTo?: string, subtractFeeFromAmount?: boolean, replaceable?: boolean) {
+    const params: any[] = [address, amount];
+
+    // Argument #6
+    if (replaceable !== undefined) {
+      params.push(comment ?? '', commentTo ?? '', subtractFeeFromAmount ?? false, replaceable);
+    }
+
+    // Argument #5
+    if (subtractFeeFromAmount !== undefined) {
+      params.push(comment ?? '', commentTo ?? '', subtractFeeFromAmount);
```

Comment:
Looks like there's potential to push too many params into the array if multiple arguments are defined? Why not just push param into array in the order seen?

`if (comment !== undefined) { params.push(comment); }` etc.

Reply:
You are right, @MrHash!